# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeParent.cpp
# Commit: f3b490c076d6
# Full Hash: f3b490c076d623d6ef3bcfd297a0522d9727477c
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-23 21:42:24
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 0c5fb3a8a94d -r f3b490c076d6 gfx/layers/wr/WebRenderBridgeParent.cpp
--- a/gfx/layers/wr/WebRenderBridgeParent.cpp	Thu Jan 23 15:35:45 2020 +0200
+++ b/gfx/layers/wr/WebRenderBridgeParent.cpp	Thu Jan 23 04:37:28 2020 +0000
@@ -1734,6 +1734,15 @@
   mApis[wr::RenderRoot::Default]->WaitFlushed();
 }
 
+void WebRenderBridgeParent::DisableNativeCompositor() {
+  // Make sure that SceneBuilder thread does not have a task.
+  mApis[wr::RenderRoot::Default]->FlushSceneBuilder();
+  // Disable WebRender's native compositor usage
+  mApis[wr::RenderRoot::Default]->EnableNativeCompositor(false);
+  // Ensure we generate and render a frame immediately.
+  ScheduleForcedGenerateFrame();
+}
+
 void WebRenderBridgeParent::UpdateQualitySettings() {
   for (auto& api : mApis) {
     if (!api) {